l5-operator/controllers/bestie_controller.go

```diff
@@ -176,17 +195,54 @@ func (r *BestieReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctr
 		}
 	}
 
-	// reconcile route
-	route := &routev1.Route{}
-
-	err = r.Get(ctx, types.NamespacedName{Name: BestieName + "-route", Namespace: bestie.Namespace}, route)
+	// Checking to see if cluster is an OpenShift cluster
+	// Checks for this api "route.openshift.io/v1"
+	isOpenShiftCluster, err := verifyOpenShiftCluster(routev1.GroupName, routev1.SchemeGroupVersion.Version)
 	if err != nil {
-		if errors.IsNotFound(err) {
-			log.Info("Creating a new route for bestie")
-			fileName := "config/resources/bestie-route.yaml"
-			r.applyManifests(ctx, req, bestie, route, fileName)
+		return ctrl.Result{}, err
+	}
+
+	// If the cluster is OpenShift, add a route, else add an ingress
```

Comment:
Could you please document which sub-path the ingress or the route will handle the network traffic? Will be installed an ingress-controller automatically? If yes, which one?